bookkeeper-server/src/main/java/org/apache/bookkeeper/client/PendingAddOp.java

```diff
@@ -284,7 +284,9 @@ public void writeComplete(int rc, long ledgerId, long entryId, BookieSocketAddre
                 // Got an error after satisfying AQ. This means we are under replicated at the create itself.
                 // Update the stat to reflect it.
                 addOpUrCounter.inc();
-                if (!lh.bk.getDisableEnsembleChangeFeature().isAvailable() && !lh.bk.delayEnsembleChange) {
+                if (!lh.bk.getDisableEnsembleChangeFeature().isAvailable()
+                        && !lh.writeFlags.contains(WriteFlag.DEFERRED_SYNC)
```

Comment:
What is the use of adding this check here?

Reply:
We are acting as "enabled" disableEnsembleChangeFeature, so I kept the same exact beheaviour

Reply:
I don't think this check makes sense. This may not cause harm as `failedBookies` set is best effort scenario.
@sijie do you think we need this check here?

Reply:
removed